### PR TITLE
Add support for pre-release

### PR DESCRIPTION
**Pull request by tagoro9**
**Description**

This PR adds 3 new actions:

* `build-docker` - Build a docker image for the project
* `build` - Run the `build` NPM script in a project - this was being done as part of release, but now we need it for prerelease too 
* `prerelease` - Trigger a release from a PR when the prerelease label is there (optionally building a docker image)

Now `release` also optionally builds docker images

This is inspired by `actions-jvm` and follows the same idea of having a local `.docker-config` file for the docker image name

**Changes**

* feat: add a build-docker action
* feat: add a build action
* refactor(release): use internal build action
* feat(release): build docker as part of release
* docs: update actions readme to rely on action docs even more
* feat: add prerelease action
* chore: use current branch for internal action refs

🚀 PR created with [fotingo](https://github.com/tagoro9/fotingo)

### CODE REVIEW

File: build-docker/action.yaml
Comment (tagoro9) [inline build-docker/action.yaml:1]: @greenkiwi It feels like this could be its own separate action. The only thing I changed was adding the npm specific inputs for tokens

File: build/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# GitHub Action Lint
```
Comment (Toolo) [inline build/README.md]: Let's reword this to build 😄

File: prerelease/action.yaml [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+    - name: Add new version to summary
+      shell: bash
+      if: steps.prerelease.outputs.new-release-published == 'true'
+      env:
+        NEW_VERSION: ${{ steps.vars.outputs.version }}
+        DOCKER_IMAGE: ${{ steps.build-docker.outputs.image-with-tag }}
+      run: |
+        echo "::notice::new version: \`${NEW_VERSION}\`"
+        echo "#### New version: \`${NEW_VERSION}\`" >> $GITHUB_STEP_SUMMARY
+        echo "#### Docekr image: \`${DOCKER_IMAGE}\`" >> $GITHUB_STEP_SUMMARY
```
Comment (Toolo) [inline prerelease/action.yaml:202]: Docker*

File: prerelease/action.yaml [hunk 3]
```diff
@@ -147,7 +147,7 @@ runs:
 
     - name: Build
       if: steps.check-pr.outputs.has_prerelease_label == 'true'
-      uses: open-turo/actions-node/build@v5
+      uses: open-turo/actions-node/build@f/prerelease
```
Comment (Toolo) [inline prerelease/action.yaml:150]: Reminder to revert this before merging 👍

File: prerelease/action.yaml [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+      uses: open-turo/actions-node/build@v5
+      with:
+        checkout-repo: false
+
+    - name: Prerelease
+      id: prerelease
+      uses: open-turo/actions-release/semantic-release@v4
+      if: steps.check-pr.outputs.has_prerelease_label == 'true'
+      with:
+        branches: '["main", {"name": "${{ steps.source-vars.outputs.branch }}","channel": "next","prerelease": "pr-${{ steps.PR.outputs.number }}.${{ github.run_number }}.${{ github.run_attempt }}"}]'
```
Comment (Toolo) [inline prerelease/action.yaml:159]: 👏 This is key to be able to release in repos with many branches

File: build-docker/action.yaml [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+      with:
+        username: ${{ inputs.dockerhub-user }}
+        password: ${{ inputs.dockerhub-password }}
+    - name: Docker meta
+      id: meta
+      uses: docker/metadata-action@v5
+      with:
+        images: |
+          ${{ steps.config.outputs.image-name }}
+        flavor: |
```
Comment (Toolo) [inline build-docker/action.yaml:80]: Can we also support customizing flavor in the docker-config file? We have an use case for it in @turo/front-end releases 💡

**Review by Toolo — APPROVED** [review]

File: build/README.md (hunk 1 above)
Comment (Toolo) [reply build/README.md]: Fixed in a later commit

File: build-docker/action.yaml (hunk 5 above)
Comment (tagoro9) [reply build-docker/action.yaml]: Added!
